src/main/java/org/springframework/hateoas/mvc/ControllerLinkBuilder.java

```diff
@@ -102,17 +105,26 @@
 	 * @param uriComponents must not be {@literal null}.
 	 */
 	ControllerLinkBuilder(UriComponents uriComponents) {
-		this(uriComponents, TemplateVariables.NONE, null);
+		this(uriComponents, TemplateVariables.NONE, null, null);
 	}
 
-	ControllerLinkBuilder(UriComponents uriComponents, TemplateVariables variables, MethodInvocation invocation) {
+	ControllerLinkBuilder(UriComponents uriComponents, TemplateVariables variables, MethodInvocation invocation, MappingDiscoverer discoverer) {
 
 		super(uriComponents);
 
 		this.variables = variables;
+		ControllerLinkBuilder.delegateDiscovererOverride = discoverer;
 		this.addAffordances(findAffordances(invocation, uriComponents));
 	}
 
+	public static void setDelegateDiscoverer(MappingDiscoverer delegateDiscovererOverride) {
```

Comment:
Some JavaDoc here would be helpful, to point out that this disables caching of `UriTemplate`s